catalog/imaging/catalog-imaging-transformer-chip/src/main/java/com/connexta/alliance/imaging/chip/transformer/CatalogInputAdapter.java

```diff
@@ -0,0 +1,74 @@
+/**
+ * Copyright (c) Connexta, LLC
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.alliance.imaging.chip.transformer;
+
+import java.io.Serializable;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import ddf.catalog.data.Attribute;
+import ddf.catalog.data.Metacard;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.impl.ResourceRequestByProductUri;
+
+/**
+ * A class to convert a Metacard to a ResourceRequest object.
+ */
+public class CatalogInputAdapter {
+
+    private static final String OVERVIEW_FRAGMENT = "overview";
+
+    /**
+     * @param metacard the Metacard containing the relevant information for the ResourceRequest.
+     *                 May not be null.
+     * @return the ResourceRequest object to query the CatalogFramework service.
+     * @throws URISyntaxException when the metacard's Resource URI isn't of the format 'content:&ltid&gt'
+     */
+    public ResourceRequest buildReadRequest(Metacard metacard) throws URISyntaxException {
+
+        if (metacard == null) {
+            throw new IllegalArgumentException("method argument 'metacard' may not be null.");
+        }
+
+        Attribute attribute = metacard.getAttribute(Metacard.RESOURCE_URI);
```

Comment:
Why not get the `DERIVED_RESOURCE_URI` and check if the fragment == `overview`. That way if none of the URI's contain `overview` then you know you can't chip this metacard's resource.